### PR TITLE
doc: Add "Improve Snapshot Ingestion" RFC.

### PR DESCRIPTION
**Pull request by breezewish**
Signed-off-by: Wish <breezewish@outlook.com>

### What problem does this PR solve?

Issue Number: ref #5237

Problem Summary:

### What is changed and how it works?

Add design doc for the Snapshot ingestion improvement.

[Rendered RFC](https://github.com/pingcap/tiflash/blob/463a879b091b3feb56c93c91be5580ca943f94d6/docs/design/2022-09-27-improve-snapshot-ingestion.md)

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: docs/design/2022-09-27-improve-snapshot-ingestion.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+   single segment.
+
+   This further improvement will be covered in future RFCs, not in this RFC.
+
+### Improvement Changes
+
+#### 1. Multi-threaded IngestSST
+
+Low Priority Apply Pool size will be increased from 1 to, for example, 4 or 8,
+so that IngestSST can be parallelized.
```
Comment (JaySon-Huang) [inline docs/design/2022-09-27-improve-snapshot-ingestion.md:176]: "which is automatically inferred from the CPU limit of tiflash"

**Review by JaySon-Huang — APPROVED** [review]
lgtm

**Review by hongyunyan — APPROVED** [review]
LGTM

**Pull request merged**